### PR TITLE
session02 homework

### PR DESCRIPTION
**Pull request by dvlupr**
Adding session01 and session02 folders
Adding four exercises
Updating readme
Adding empty test file

### CODE REVIEW

File: students/JimJenkins/session01/break_me.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+Author: Jim Jenkins (dvlupr)
+Date: 10/13/2018
+
+Each function, when called, should cause an exception to happen
+Each function should result in one of the four most common exceptions you’ll find.
+For review: NameError, TypeError, SyntaxError, AttributeError
+
+'''
+# functions that trigger each error
+def fn_nameerror():
```
Comment (charlesrobison) [inline students/JimJenkins/session01/break_me.py]: Typically you'll want your function names to be in snake_case.  def name_error(): for example.

File: students/JimJenkins/session01/break_me.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    print nameerror
```
Comment (charlesrobison) [inline students/JimJenkins/session01/break_me.py]: Missing the parentheses and quotes on this one - I'm getting a SyntaxError

File: students/JimJenkins/session01/break_me.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+def fn_typeerror():
+    print('typeerror')
```
Comment (charlesrobison) [inline students/JimJenkins/session01/break_me.py]: For this, you'd want to create a function that actually produces the error.  Counterintuitive, I know :)

File: students/JimJenkins/session01/break_me.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+def fn_syntaxerror():
+    print('syntaxerror
+
+def fn_attributeError():
+    print('attributeerror')
```
Comment (charlesrobison) [inline students/JimJenkins/session01/break_me.py]: Same thing on these - camel_case and produce actual errors.  Take a look and reach out if you have questions.

File: students/JimJenkins/session02/fizzbuzz.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    # check to ensure the number of sequence runs is reasonable
+    if y > 500:
+        sys.exit('Number is too high for the system, please select a lower number')
+    x = x + 1
+    if x % 3 == 0:
+        print (x, 'Fizz')
+    elif x % 5 == 0:
+        print(x, 'Buzz')
+    else:
+        print(x, 'FizzBuzz')
```
Comment (charlesrobison) [inline students/JimJenkins/session02/fizzbuzz.py]: This is pretty close - but not quite.  Getting a lot more FizzBuzz that we should be.  Can you think of a way to simplify the logic a bit?

File: students/JimJenkins/session02/grid_printer.py [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+            print(('|' + ('  ' * x) + ' ') * a + '|')
+    print(('+' + (' -' * x) + ' ') * a + '+')
+
+
+# format the view
+print('-----' * 7)
+
+
+# run the fundtion based on user input
+fn_grid(x,a)
```
Comment (charlesrobison) [inline students/JimJenkins/session02/grid_printer.py]: Works well - nicely done

File: students/JimJenkins/session02/series.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        z = x + y
+        # add the item to the list
+        num_list.append(z)
+        print(num_list)
+
+# ask the user for how many times they want to run the sequence
+b = int(input('How many times would you like to run the sequence?: '))
+
+# run the sequence per the users instructions
+fn_fibonacci(b)
```
Comment (charlesrobison) [inline students/JimJenkins/session02/series.py]: Looks like this one's not quite finished - still working on lucas, sum_series, and tests?

File: students/JimJenkins/session01/break_me.py (hunk 2 above)
Comment (PythonCHB) [reply students/JimJenkins/session01/break_me.py]: NOTE: ifthis code runs, then you are running it in Python2 -- you really do want to figure out how to get Python3 running -- they aren't all that different, but we do use Python3 only features in this class.

File: students/JimJenkins/session02/series.py (hunk 7 above)
Comment (PythonCHB) [reply students/JimJenkins/session02/series.py]: and we're looking for a function that returns the nth value -- not the whole series ....

and try to do it using recursion.

File: students/JimJenkins/session02/series.py [hunk 8]
```diff
@@ -1,39 +1,58 @@
...
-        z = x + y
-        # add the item to the list
-        num_list.append(z)
-        print(num_list)
 
-# ask the user for how many times they want to run the sequence
-b = int(input('How many times would you like to run the sequence?: '))
+def sum_series(n, x=0, y=1):
+    if n == 0:
+        return n == x
```
Comment (PythonCHB) [inline students/JimJenkins/session02/series.py]: This is going to return “True” or “False”.

File: students/JimJenkins/session02/fizzbuzz.py [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+# request how many times to run FizzBuzz
+y = 100
+
+
+# set the sentinel value
+x = 0
+
+
+# create a while loop looking for the factors of 3 and 5
+while x != y:
```
Comment (PythonCHB) [inline students/JimJenkins/session02/fizzbuzz.py]: A for loop with range() is a better option here.

File: students/JimJenkins/session02/fizzbuzz.py [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+    x = x + 1
+    if x % 3 == 0 and x % 5 == 0:
+        print(x, 'FizzBuzz')
+    elif x % 3 == 0:
+        print(x, 'Fizz')
+    elif x % 5 == 0:
+        print(x, 'Buzz')
+    else:
+        print(x)
```
Comment (PythonCHB) [inline students/JimJenkins/session02/fizzbuzz.py]: Logic all good!

File: students/JimJenkins/session02/series.py [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+
+
+def lucas(n):
+    if n == 0:
+        return 2
+    elif n == 1:
+        return 1
+    else:
+        return lucas(n-1)+lucas(n-2)
+
```
Comment (PythonCHB) [inline students/JimJenkins/session02/series.py]: Fib and Lucas are perfect :-)

File: students/JimJenkins/session02/series.py [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+def sum_series(n, x=0, y=1):
+    if n == 0:
+        return n == x
+    elif n == 1:
+        return n == y
+    else:
+        return sum_series(n-1)+sum_series(n-2)
```
Comment (PythonCHB) [inline students/JimJenkins/session02/series.py]: You need to
Pass x and y through, or you will get the Fibonacci series every time ..,.

File: students/JimJenkins/session02/series.py (hunk 11 above)
Comment (dvlupr) [reply students/JimJenkins/session02/series.py]: updated sum_series  
added assertion

**Comment by PythonCHB** [comment]
Merging this baby!

**Pull request merged**